frame/fast-unstake/src/tests.rs

```diff
@@ -990,8 +944,255 @@ mod on_idle {
 			assert_eq!(
 				fast_unstake_events_since_last_call(),
 				vec![
-					Event::Checking { stash: 42, eras: vec![3, 2, 1, 0] },
-					Event::Unstaked { stash: 42, result: Ok(()) }
+					Event::BatchChecked { eras: vec![3, 2, 1, 0] },
+					Event::Unstaked { stash: 42, result: Ok(()) },
+					Event::BatchFinished
+				]
+			);
+		});
+	}
+}
+
+mod batched {
+	use super::*;
+
+	#[test]
+	fn single_block_batched_successful() {
+		ExtBuilder::default().batch(3).build_and_execute(|| {
+			ErasToCheckPerBlock::<T>::put(BondingDuration::get() + 1);
+			CurrentEra::<T>::put(BondingDuration::get());
+
+			assert_ok!(FastUnstake::register_fast_unstake(RuntimeOrigin::signed(2)));
+			assert_ok!(FastUnstake::register_fast_unstake(RuntimeOrigin::signed(4)));
+			assert_ok!(FastUnstake::register_fast_unstake(RuntimeOrigin::signed(6)));
+			assert_ok!(FastUnstake::register_fast_unstake(RuntimeOrigin::signed(8)));
+
+			assert_eq!(Queue::<T>::count(), 4);
+			assert_eq!(Head::<T>::get(), None);
+
+			// when
+			next_block(true);
+
+			// then
+			assert_eq!(
+				Head::<T>::get(),
+				Some(UnstakeRequest {
+					stashes: bounded_vec![
```

Comment:
`BatchSize` is only `1` so I am confused as to why 3 out of the 4 stashes are put into the `Head` here?